### PR TITLE
refactor(components): menu migration to fast

### PR DESCRIPTION
**Pull request by yinonov**

### CODE REVIEW

**Review by vng-vivid — APPROVED** [review]

File: libs/components/src/lib/popup/popup.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export class Popup extends FoundationElement {
 	 * @public
 	 * HTML Attribute: placement
 	 */
-	@attr({ mode: 'fromView' }) placement?: Placement = 'left';
+	@attr({ mode: 'fromView' }) placement?: Placement;
```
Comment (rinaok) [inline libs/components/src/lib/popup/popup.ts:77]: You have to change it also in the readme file

Comment (rinaok) [inline libs/components/src/lib/popup/popup.ts:77]: Should be changed also in the Popup's readme file

File: libs/components/src/lib/menu/ui.test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import type { Page } from '@playwright/test';
+import {
+	extractHTMLBlocksFromReadme,
+	loadComponents,
+	loadTemplate,
+} from '../../visual-tests/visual-tests-utils.js';
+
+const components = ['menu'];
+
+test('should show the component', async ({ page }: { page: Page }) => {
```
Comment (rinaok) [inline libs/components/src/lib/menu/ui.test.ts:12]: The ui tests don't look good. See Popup's ui tests for reference.

File: libs/components/src/lib/menu/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+## Methods
+
+### focus
+
+- Returns: void
+
+Focuses the first item in the menu.
+
+### collapseExpandedItem
```
Comment (rinaok) [inline libs/components/src/lib/menu/README.md:73]: misses test

File: libs/components/src/lib/menu/menu.template.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ * @public
+ */
+export const MenuTemplate: (
+	context: ElementDefinitionContext,
+	definition: FoundationElementDefinition
+) => ElementViewTemplate = (context: ElementDefinitionContext) => {
+	const popupTag = context.tagFor(Popup);
+
+	return html<Menu>`
+		<template
```
Comment (rinaok) [inline libs/components/src/lib/menu/menu.template.ts:27]: why not `<menu>`?

File: libs/components/src/lib/checkbox/index.ts [hunk 5]
```diff
@@ -11,10 +11,7 @@ import { CheckboxTemplate as template } from './checkbox.template';
 export const vividCheckbox = Checkbox.compose<CheckboxOptions>({
 	baseName: 'checkbox',
 	template: template as any,
-	styles,
-	shadowOptions: {
-		delegatesFocus: true,
-	}
+	styles
```
Comment (rinaok) [inline libs/components/src/lib/checkbox/index.ts:14]: not connected to this PR

File: libs/components/src/lib/menu/ui.test.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import * as path from 'path';
+import { expect, test } from '@playwright/test';
+import type { Page } from '@playwright/test';
+import {
+	extractHTMLBlocksFromReadme,
+	loadComponents,
+	loadTemplate,
+} from '../../visual-tests/visual-tests-utils.js';
+
+const components = ['menu'];
```
Comment (rinaok) [inline libs/components/src/lib/menu/ui.test.ts:10]: use also use button in the ui test

File: README.md [hunk 7]
```diff
@@ -52,7 +52,7 @@ Feel free to check [issues page](https://github.com/Vonage/vivid-3/issues). You
 
 #### Generator a new component
 
-`npx run nx g @vonage/nx-vivid:component my-component`
```
Comment (rinaok) [inline README.md:55]: not connected to this PR

**Comment by rinaok** [comment]
@yinonov Even tough you've already merged it but I have some comments

File: libs/components/src/lib/menu/ui.test.ts (hunk 2 above)
Comment (yinonov) [reply libs/components/src/lib/menu/ui.test.ts:12]: there's no proper design as items aren't available

File: libs/components/src/lib/popup/popup.ts (hunk 1 above)
Comment (yinonov) [reply libs/components/src/lib/popup/popup.ts:77]: is it documented that the default is `left`?

File: libs/components/src/lib/menu/menu.template.ts (hunk 4 above)
Comment (yinonov) [reply libs/components/src/lib/menu/menu.template.ts:27]: native menu tag? isn't advised or adds any value

File: README.md (hunk 7 above)
Comment (yinonov) [reply README.md:55]: true, noticed the typo when tried to generate new component

File: libs/components/src/lib/menu/ui.test.ts (hunk 6 above)
Comment (yinonov) [reply libs/components/src/lib/menu/ui.test.ts:10]: addressed at #625

File: libs/components/src/lib/checkbox/index.ts (hunk 5 above)
Comment (yinonov) [reply libs/components/src/lib/checkbox/index.ts:14]: true 🫢

File: libs/components/src/lib/menu/menu.template.ts (hunk 4 above)
Comment (yinonov) [reply libs/components/src/lib/menu/menu.template.ts:27]: didn't get it

File: libs/components/src/lib/menu/README.md (hunk 3 above)
Comment (yinonov) [reply libs/components/src/lib/menu/README.md:73]: this is part of the super class scope but if we reflect that in docs, it's probably correct to test.
problem here is items are not yet available to use in test. will have to keep in mind

**Pull request merged**